src/test/mockFactory.ts

```diff
@@ -0,0 +1,80 @@
+import Serverless from 'serverless';
+import Service from 'serverless/classes/Service';
+import Utils = require('serverless/classes/Utils');
+import PluginManager = require('serverless/classes/PluginManager');
+
+export class MockFactory {
+  public static createTestServerless(service?: Service, utils?: Utils, pluginManager?: PluginManager): Serverless {
```

Comment:
Do you see a use case where we would want to manually create `utils` or `pluginManager`?  I would expect if we wanted to mock any of those calls we would just spy/mock on them after instantiation.